internal/codegen/golang/postgresql_type.go

```diff
@@ -246,23 +245,26 @@ func postgresType(r *compiler.Result, col *compiler.Column, settings config.Comb
 			return "interface{}"
 		}
 		if rel.Schema == "" {
-			rel.Schema = r.Catalog.DefaultSchema
+			rel.Schema = req.Catalog.DefaultSchema
 		}
 
-		for _, schema := range r.Catalog.Schemas {
+		for _, schema := range req.Catalog.Schemas {
 			if schema.Name == "pg_catalog" {
 				continue
 			}
-			for _, typ := range schema.Types {
-				switch t := typ.(type) {
-				case *catalog.Enum:
-					if rel.Name == t.Name && rel.Schema == schema.Name {
-						if schema.Name == r.Catalog.DefaultSchema {
-							return StructName(t.Name, settings)
-						}
-						return StructName(schema.Name+"_"+t.Name, settings)
+
+			for _, enum := range schema.Enums {
+				if rel.Name == enum.Name && rel.Schema == schema.Name {
+					if schema.Name == req.Catalog.DefaultSchema {
+						return StructName(enum.Name, req.Settings)
 					}
-				case *catalog.CompositeType:
+					return StructName(schema.Name+"_"+enum.Name, req.Settings)
+				}
+			}
+
+			for _, ct := range schema.CompositeTypes {
+				// XXX: old code had a bug?
```

Comment:
i think the old code... did not check that the name/schema actually matched? but this is conjecture and i did not try to write a test